cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -262,6 +264,12 @@ public final ProgramController run(final Program program, ProgramOptions oldOpti
           }
           twillPreparer.setLogLevels(runnable, transformLogLevels(runnableDefinition.getLogLevels()));
           twillPreparer.withConfiguration(runnable, runnableDefinition.getTwillRunnableConfigs());
+          if (clusterMode == ClusterMode.ON_PREMISE && program.getType() == ProgramType.WORKFLOW) {
+            // set TwillConstants#PROGRAM_RUNTIME_ENV configuration on twill preparer. This configuration will be
+            // used when cdap programs are launched on k8s env.
+            twillPreparer.withConfiguration(runnable,
```

Comment:
I think configuration on the TwillPreparer should be moved into the `ProgramLaunchConfig` class such that individual ProgramRunner implementation can configure it accordingly.

Also, why the constant is called PROGRA_RUNTIME_ENV and the value as "master"?? Seems unclear what does it mean.

Reply:
1. makes sense, moving to `ProgramLaunchConfig`
2. For naming, I set it to that for lack of better word in the context of program and master services run in same env. Though you are right, it does look little strange. 
May using `native` Or `onprem` env is a better value for this property?

Reply:
I am using `onpremise` for now. Please let me know if you have any other suggestions.